app/scripts/model/dataProxy.js

```diff
@@ -1082,37 +1095,34 @@ window.DataManagerForIviz = (function($, _) {
         }),
       getCaseLists: window.cbio.util.makeCachedPromiseFunction(
         function(self, fetch_promise) {
-          var _allSampleIds = [];
-          var _sequencedSampleIds = [];
-          var _cnaSampleIds = [];
+          var _responseStudyCaseList = {};
           var requests = self.getCancerStudyIds().map(
             function(cancer_study_id) {
               var def = new $.Deferred();
-              var sampleListCalls = [];
               self.getSampleListsData(['all', 'sequenced', 'cna'], cancer_study_id)
                 .done(function() {
+                  var studyCaseList = {sequencedSampleIds:[],
+                    cnaSampleIds: [],
+                    allSampleIds: []
+                  };
                   if (_.isArray(self.data.sampleLists.sequenced[cancer_study_id])) {
-                    _sequencedSampleIds = _sequencedSampleIds.concat(self.data.sampleLists.sequenced[cancer_study_id]);
+                    studyCaseList.sequencedSampleIds  = self.data.sampleLists.sequenced[cancer_study_id];
                   }
                   if (_.isArray(self.data.sampleLists.cna[cancer_study_id])) {
-                    _cnaSampleIds = _cnaSampleIds.concat(self.data.sampleLists.cna[cancer_study_id]);
+                    studyCaseList.cnaSampleIds = self.data.sampleLists.cna[cancer_study_id];
                   }
                   if (_.isArray(self.data.sampleLists.all[cancer_study_id])) {
-                    _allSampleIds = _allSampleIds.concat(self.data.sampleLists.all[cancer_study_id]);
+                    studyCaseList.allSampleIds = self.data.sampleLists.all[cancer_study_id];
```

Comment:
This needs to be sorted. Please check left panel to make sure all lists are sorted